spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/FindIncreasedAccessibilityOfMethodsTest.java

```diff
@@ -0,0 +1,121 @@
+package edu.umd.cs.findbugs.detect;
+
+import edu.umd.cs.findbugs.AbstractIntegrationTest;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcher;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcherBuilder;
+import org.junit.Test;
+
+import static edu.umd.cs.findbugs.test.CountMatcher.containsExactly;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.not;
+
+public class FindIncreasedAccessibilityOfMethodsTest extends AbstractIntegrationTest {
+
+    @Test
+    public void findIAOMBugInClass_IncreasedAccessibilityOfMethods_SubClassFromSamePackage() {
+        performAnalysis(
+                "increasedAccessibilityOfMethods/SuperClassOfSuperClass.class",
+                "increasedAccessibilityOfMethods/SuperClass.class",
+                "increasedAccessibilityOfMethods/SubClassFromSamePackage.class");
+
+        assertContainsCorrectNumberOfIAOMBugs(6);
+
+        assertContainsIAOMBug("SubClassFromSamePackage", "protectedMethodToPublic");
+        assertContainsIAOMBug("SubClassFromSamePackage", "packagePrivateMethodToPublic");
+        assertContainsIAOMBug("SubClassFromSamePackage", "packagePrivateMethodToProtected");
+        assertContainsIAOMBug("SubClassFromSamePackage", "superProtectedMethodToPublic");
+        assertContainsIAOMBug("SubClassFromSamePackage", "superPackagePrivateMethodToPublic");
+        assertContainsIAOMBug("SubClassFromSamePackage", "superPackagePrivateMethodToProtected");
+    }
+
+    @Test
+    public void findIAOMBugInClass_IncreasedAccessibilityOfMethods_SubClassFromAnotherPackage() {
+        performAnalysis(
+                "increasedAccessibilityOfMethods/SuperClassOfSuperClass.class",
+                "increasedAccessibilityOfMethods/SuperClass.class",
+                "increasedAccessibilityOfMethods/anotherPackage/SubClassFromAnotherPackage.class");
+
+        assertContainsCorrectNumberOfIAOMBugs(1);
+
+        assertContainsIAOMBug("SubClassFromAnotherPackage", "protectedMethodToPublic");
+    }
+
+    @Test
+    public void findNoIAOMBugInClass_IncreasedAccessibilityOfMethods_CorrectSubClass() {
+        performAnalysis(
+                "increasedAccessibilityOfMethods/SuperClassOfSuperClass.class",
+                "increasedAccessibilityOfMethods/anotherPackage/CorrectSubClass.class");
+
+        assertContainsCorrectNumberOfIAOMBugs(0);
+    }
+
+    @Test
+    public void findNoIAOMBugInClass_IncreasedAccessibilityOfMethods_CloneableImplementation() {
+        performAnalysis(
+                "increasedAccessibilityOfMethods/CloneableImplementation.class");
+
+        assertContainsCorrectNumberOfIAOMBugs(0);
+    }
+
+    @Test
+    public void findNoIAOMBugInClass_IncreasedAccessibilityOfMethods_InterfaceImplementation() {
+        performAnalysis(
+                "increasedAccessibilityOfMethods/InterfaceImplementation.class");
+
+        assertContainsCorrectNumberOfIAOMBugs(0);
```

Comment:
Yeah, I think it should not. The best practice with interfaces is to don't use accessibility modifiers - not because package private accessibility is important. but because it doesn't matter. I suspect, that you are not sharing my idea.

Reply:
Thanks, yes, you are right.